### DISCUSSION TITLE
No consumers for a Kafka topic

### DISCUSSION THREAD

**Discussion by hariraogotit** [spring-projects/spring-kafka#2819] [Q&A] [answered]
Team,

  When I run a command using kafka-consumer-groups.sh to list  the current offeset, lag etc for a group, I see no consumer-id attached although there is a lag in a topic. The topic is listened by a service and service is up and running. 

When I restart the service, the consumer is attached and processed the messages for 7 mins then again the consumer-id became empty and no messages are processed. There are 500 more messages that are yet to be processed

I do not see any error in the broker.

I have the following config

```
max-poll-interval: 600000
 max-poll-records: 1000
```

So why does the consumer detaching from the topic?

Any leads are much appreciated and let me know if you need more info.

**Comment by artembilan** [comment, marked as answer]
> So why does the consumer detaching from the topic?

Apparently you have some blocking part in your listener logic where you don't return control back to `KafkaConsumer`.
And since it does not `poll()` any more, the group coordinator treats it as broken and detaches it from the group.

Please, revise the logic of your `@KafkaListener` and try to determine where you block.

**Reply by hariraogotit** [reply]
Thank you. Will check but no errors/warnings in broker is concerning.

**Reply by hariraogotit** [reply]
Yes one of the down stream is taking more time to respond.
